ui/app/AppLayouts/Profile/views/NotificationsView.qml

```diff
@@ -1,399 +1,576 @@
+import StatusQ.Components 0.1
+import StatusQ.Popups 0.1
 
 import utils 1.0
-
 import shared.panels 1.0
-import shared.status 1.0
 import shared.controls 1.0
 
 import "../stores"
-import "../popups"
+import "../controls"
 import "../panels"
-import "./"
+import "../popups"
 
-ScrollView {
+Item {
     id: root
-
     property NotificationsStore notificationsStore
-
+    property DevicesStore devicesStore
     property int profileContentWidth
 
-    height: parent.height
-    width: parent.width
-    contentHeight: notificationsContainer.height
-    clip: true
+    ButtonGroup {
+        id: messageSetting
+    }
 
-    Item {
-        id: notificationsContainer
-        width: profileContentWidth
-        anchors.horizontalCenter: parent.horizontalCenter
-        height: this.childrenRect.height + 100
+    Loader {
+        id: exemptionNotificationsModal
+        active: false
 
-        property Component mutedChatsModalComponent: MutedChatsModal {}
+        function open(item) {
+            active = true
+            exemptionNotificationsModal.item.item = item
+            exemptionNotificationsModal.item.open()
+        }
+        function close() {
+            active = false
+        }
 
+        sourceComponent: ExemptionNotificationsModal {
+            anchors.centerIn: parent
+            notificationsStore: root.notificationsStore
 
-        ButtonGroup {
-            id: notificationSetting
+            onClosed: {
+                exemptionNotificationsModal.close();
+            }
         }
+    }
 
-        ButtonGroup {
-            id: soundSetting
-        }
+    Component {
+        id: exemptionDelegateComponent
+        StatusListItem {
+            property string lowerCaseSearchString: searchBox.text.toLowerCase()
 
-        ButtonGroup {
-            id: messageSetting
-        }
+            width: parent.width
+            height: visible ? implicitHeight : 0
+            visible: lowerCaseSearchString === "" ||
+                     model.itemId.toLowerCase().includes(lowerCaseSearchString) ||
+                     model.name.toLowerCase().includes(lowerCaseSearchString)
+            title: model.name
+            subTitle: {
+                if(model.type === Constants.settingsSection.exemptions.community)
+                    return qsTr("Community")
+                else if(model.type === Constants.settingsSection.exemptions.oneToOneChat)
+                    return qsTr("1:1 Chat")
+                else if(model.type === Constants.settingsSection.exemptions.groupChat)
+                    return qsTr("Group Chat")
+                else
+                    return ""
+            }
+            label: {
+                if(!model.customized)
+                    return ""
+
+                let l = ""
+                if(model.muteAllMessages)
+                    l += qsTr("Muted")
+                else {
+                    let singleChange = false
+                    let multipleChanges = false
```

Comment:
Maybe instead of single and multiple changes, you could use an int called `nbOfChanges` and increment it each time there is a change. That way, in the end, when you detect if there are multiple changes, just check if `nbOfChanges > 1`

Reply:
@jrainville yes, that's true, simpler, updated (initially I had some other logic there and just continued in that way).